init.lua

```diff
@@ -742,28 +759,50 @@ local craftable_setting = minetest.settings:get_bool("personal_log_craftable_ite
 
 if craftable_setting or not (unified_inventory_modpath or sfinv_modpath or sfinv_buttons_modpath) then
 
-minetest.register_craftitem("personal_log:book", {
-	description = S("Personal Log"),
-	inventory_image = "personal_log_open_book.png",
-	groups = {book = 1, flammable = 3},
-	on_use = function(itemstack, user, pointed_thing)
-		local name = user:get_player_name()
-		minetest.show_formspec(name,"personal_log:root", make_personal_log_formspec(user))
-	end,
-})
+	attributes = {
+		description = S("Personal Log"),
+		inventory_image = "personal_log_open_book.png",
+		groups = {book = 1, flammable = 3},
+	}
 
-minetest.register_craft({
-	output = "personal_log:book",
-	recipe = {{book_unwritten, book_unwritten}}
-})
+	if mcl_formspec_modpath then
+		attributes.on_secondary_use = function(itemstack, user, pointed_thing)
+			local name = user:get_player_name()
+			minetest.show_formspec(name,"personal_log:root", make_personal_log_formspec(user))
+		end
+		attributes.on_place = function(itemstack, user, pointed_thing)
+			if not user:get_player_control().sneak then 
+				local new_stack = mcl_util.call_on_rightclick(itemstack, user, pointed_thing)
+				if new_stack then
+					return new_stack
+				end
+			end
+			local name = user:get_player_name()
+			minetest.show_formspec(name,"personal_log:root", make_personal_log_formspec(user))
+		end
+	else
+		attributes.on_use = function(itemstack, user, pointed_thing)
+			if mcl_formspec_modpath then return nil end
```

Comment:
What's the idea behind this? You should opt-depend on this mod to make sure the presence checks are working correctly (if not already done). This line is unnecessary if things are implemented correctly.

Reply:
I did this so that it would be used on right-click in MineClone and left click in other games.

Reply:
I didn't see that; fixed.